src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -144,24 +149,39 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       return;
     }
 
+    // this code uses markdown (flexmark library) to parse to html to show on conversation chat
+    // Jsoup library is used to clean out any script and unwanted html tags
+
+    // use Jsoup to allow certain tags for parsing
+    Whitelist allowedTags = Whitelist.none(); //no tags allowed, empty whitelist
+    // now add tags to empty whitelist
+    // ins: underline, sub: subscript, strong: bold, em: italics, del: strikethrough, sup: superscript
+    allowedTags.addTags("ins", "sub", "strong", "em", "del", "sup");
+
+    // this allows for extensions to be added for markdown
+    MutableDataSet options = new MutableDataSet();
+
+    // set underline(++) and strikethrough(~~) extension for markdown
+    options.set(Parser.EXTENSIONS, Arrays.asList(InsExtension.create(), StrikethroughExtension.create()));
+
+    // parse markdown to html
+    Parser parser = Parser.builder(options).build();
+    HtmlRenderer renderer = HtmlRenderer.builder(options).build();
+
+    // re-use parser and renderer instancess
     String messageContent = request.getParameter("message");
+    Node document  = parser.parse(messageContent);
+    String markdownContent = renderer.render(document);
+
+    // this deletes new line tag that parse auto creates at end of node
+    String markdownContentNoNewLine = markdownContent.replaceAll("\n","");
```

Comment:
A whitespace between `,` & `""`. The autoformatter might help with this (mentioned in an earlier comment)